bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/type/ChannelType.java

```diff
@@ -63,16 +66,56 @@
      *             or the the meta information is null
      */
     public ChannelType(ChannelTypeUID uid, boolean advanced, String itemType, String label, String description,
-            String category, Set<String> tags, StateDescription state, URI configDescriptionURI)
-                    throws IllegalArgumentException {
-
-        super(uid, label, description);
+            String category, Set<String> tags, StateDescription state, URI configDescriptionURI) {
+        this(uid, advanced, itemType, null, label, description, category, tags, state, null, configDescriptionURI);
 
         if ((itemType == null) || (itemType.isEmpty())) {
             throw new IllegalArgumentException("The item type must neither be null nor empty!");
         }
+    }
+
+    /**
+     * Creates a new instance of this class with the specified parameters.
+     *
+     * @param uid the unique identifier which identifies this Channel type within
+     *            the overall system (must neither be null, nor empty)
+     *
+     * @param advanced true if this channel type contains advanced features, otherwise false
+     *
+     * @param itemType the item type of this Channel type, e.g. {@code ColorItem}
+     *
+     * @param triggerType the trigger type of this Channel type
+     *
+     * @param label the human readable label for the according type
+     *            (must neither be null nor empty)
+     *
+     * @param description the human readable description for the according type
+     *            (could be null or empty)
+     *
+     * @param category the category of this Channel type, e.g. {@code TEMPERATURE} (could be null or empty)
+     *
+     * @param tags all tags of this {@link ChannelType}, e.g. {@code Alarm} (could be null or empty)
+     *
+     * @param state the restrictions of an item state which gives information how to interpret it
+     *            (could be null)
+     *
+     * @param configDescriptionURI the link to the concrete ConfigDescription (could be null)
+     *
+     * @throws IllegalArgumentException if the UID or the item type is null or empty,
+     *             or the the meta information is null
+     */
+    public ChannelType(ChannelTypeUID uid, boolean advanced, String itemType, String triggerType, String label,
```

Comment:
ChannelType has two constructors: The first one for use without triggerType, the second for use withTriggerType (itemType is then null). I can't add a constructor only for triggerType, as itemType and triggerType are both `String` and then it would clash with the `itemType` constructor.